internal/planning/deploy/deploy.go

```diff
@@ -782,12 +779,10 @@ func prepareRunOpts(ctx context.Context, stack *planning.Stack, srv planning.Ser
 		return err
 	}
 
-	if merged.WorkingDir != "" {
-		out.MainContainer.WorkingDir = merged.WorkingDir
+	err = out.MainContainer.BinaryConfig.MergeFrom(merged)
+	if err != nil {
```

Comment:
Same here, (a) lets define our own merge, and (b) avoid re-using variables in Go (can easily use if statement scoping).